src/proxy/momento/src/protocol/resp/zadd.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2025 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::io::Write;
+use std::time::Duration;
+
+use momento::cache::SortedSetElement;
+use momento::CacheClient;
+use protocol_resp::{SortedSetAdd, SortedSetIncrement, ZADD, ZADD_EX};
+use tokio::time;
+
+use crate::error::ProxyResult;
+use crate::klog::{klog_1, Status};
+use crate::ProxyError;
+
+use super::{parse_sorted_set_score, update_method_metrics, zincrby};
+
+pub async fn zadd(
+    client: &mut CacheClient,
+    cache_name: &str,
+    response_buf: &mut Vec<u8>,
+    req: &SortedSetAdd,
+) -> ProxyResult {
+    update_method_metrics(&ZADD, &ZADD_EX, async move {
+        let number_of_elements_added = req.members().len();
+
+        // Momento does not yet support some of these optional arguments, return an error if any are set
+        if req.optional_args().ch
+            || req.optional_args().xx
+            || req.optional_args().nx
+            || req.optional_args().gt
+            || req.optional_args().lt
+        {
+            klog_1(&"zadd", &req.key(), Status::ServerError, 0);
+            return Err(ProxyError::from(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Momento proxy does not support CH, XX, NX, GT, or LT optional arguments",
+            )));
+        }
+
+        // If INCR is set, then ZADD should behave like ZINCRBY (as per the docs), which accepts only a single score-member pair
+        if req.optional_args().incr {
+            if req.members().len() != 1 {
+                klog_1(&"zadd", &req.key(), Status::ServerError, 0);
+                return Err(ProxyError::from(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    "INCR option requires exactly one score-member pair",
+                )));
```

Comment:
This logic should be in the request parsing (ie any request like this should be malformed, whether it is going to a Momento backend or any other backend).